CleverRaven/Cataclysm-DDA
# [CR] Pathfinding accounts for SMALL_PASSAGE

#### Summary
Bugfixes "Pathfinding accounts for SMALL_PASSAGE"


#### Purpose of change
Fixes #77385


#### Describe the solution
* Sets the so far-unused PathfindingFlag::RestrictLarge and PathfindingFlag::RestrictHuge flags, added in #76026, for SMALL_PASSAGE terrain
* Check for that flag when calculating pathfinding costs to make the terrain count as impassable
* Update pathfinding settings to account for size
* Make character's settings update as their size category changes.

I added the check for openable but impassable terrain as its own if block instead of trying to integrate it into the already existing one below it for sake of readability.

TODO before undrafting
- [x] Make the struct member nullable or something so that it doesn't need to be checked for pathfinding that doesn't need to know about character size, particularly for dropping items on the floor
- [x] Figure out how to handle closed windows, which the pathfinder sees as openable, but doesn't know they're not necessarily passable once open


#### Describe alternatives you've considered



#### Testing
Tested with repro save from #77385:
0. Load repro save
1 . `O` -> `Sort out my loot (static zones only)`
2. Observe that character will not try to open windows or pass through window frames
3. Load save again
4. Debug edit away the "Incoveniently large" mutation
5. Observe that character now paths through window frames and opens windows from the inside while looting progresses


#### Additional context
Draft for 0.H version of the fix #77550
